### PR TITLE
Feature/ntgl 384 multiple timetables

### PR DESCRIPTION
**Pull request by Shzmj**
- Users can create and work with Multiple timetables!

image

Co-authored-by: @M7s7 (Michael Siu) and @eklavya-joshi (Eklavya Joshi)

### CODE REVIEW

**Comment by manhualu** [comment]
- Can we make the dialog for clear current or all timetable modal similar to when we make changes without saving the changes to events?

File: client/src/App.tsx [hunk 1]
```diff
@@ -106,10 +112,10 @@ const App: React.FC = () => {
   setDropzoneRange(days.length, earliestStartTime, latestEndTime);
 
   /**
-   * Attemps callback() several times before raising error. Intended for unreliable fetches
+   * Attempts callback() several times before raising error. Intended for unreliable fetches
    */
   const maxFetchAttempts: number = 6;
-  const fetchCooldown: number = 120; // milliseconds
+  const fetchCooldown: number = 120; //
```
Comment (Rayahhhmed) [inline client/src/App.tsx]: Very interesting comment. Could you fix this and put the units of time asw

File: client/src/App.tsx [hunk 2]
```diff
@@ -269,6 +284,23 @@ const App: React.FC = () => {
   const handleRemoveCourse = (courseCode: CourseCode) => {
     const newSelectedCourses = selectedCourses.filter((course) => course.code !== courseCode);
     setSelectedCourses(newSelectedCourses);
+    const newCourseData = courseData;
+    newCourseData.map = courseData.map.filter((targetCourse) => {
+      for (const timetable of displayTimetables) {
+        for (const course of timetable.selectedCourses) {
+          console.log(course.code + targetCourse.code);
```
Comment (Rayahhhmed) [inline client/src/App.tsx]: Remove console log here

File: client/src/App.tsx [hunk 3]
```diff
@@ -269,6 +284,23 @@ const App: React.FC = () => {
...
+          if (course.code.localeCompare(courseCode)) {
+            return true;
+          }
+        } 
+      }
+      console.log("removed" + courseCode);
+      return false;
+    });
+    setCourseData(newCourseData);
+    // for (const timetable in displayTimetables) {
```
Comment (Rayahhhmed) [inline client/src/App.tsx]: Remove this comment as well

**Review by Rayahhhmed — APPROVED** [review]

**Review by manhualu — APPROVED** [review]

**Pull request merged**